splice_machine/src/main/java/com/splicemachine/derby/stream/ActivationHolder.java

```diff
@@ -130,7 +129,10 @@ private void addSubOperations(Map<Integer, SpliceOperation> operationsMap, Splic
     }
 
     public Activation getActivation() {
-        init();
+        // Reinitialize if deserialized
+        if (!initialized) {
+            init(txn, true);
+        }
         return activation;
```

Comment:
If we're returning the activation outside of synchronization, is there a chance some other thread is updating it (e.g. a call to reinitialize), and we could get a partially updated copy?  General question:  Will each thread deserialize its own ActivationHolder objects, or could these be shared?

Reply:
I agree, activation should not be read outside the synchronized block.

Reply:
I did some reading and apparently assignment and reading of references is atomic in Java:[https://stackoverflow.com/questions/44283378/why-is-reference-assignment-atomic-in-java/44284401](https://stackoverflow.com/questions/44283378/why-is-reference-assignment-atomic-in-java/44284401).  So, it sounds like we won't ever see a reference which is partially updated.  But I'm wondering if in the future activation should be made thread local, or use some copy-on-write solution.  I've seen places in the code where elements of the activation are updated at runtime.